### PR TITLE
Test remotes picked when doing a install with --update

### LINKED ISSUE #6104: [question] Which remote is picked up by conan install --update
We use fmt/6.0.0@_/_. We have a (older) version of this recipe on our local conan server with all the prebuilt packages we need for our CI. The CI server configurations declares our conan server as a remote first and then conan-center, bincrafters, etc.. Finally, they use `conan install path/to/file --update --build never --profile gcc8-release`.

This sometimes failed because conan complains about `fmt` missing its prebuilt package. It appears that conan have updated the `fmt` recipe and picked the one from conan-center and NOT the one on our server. I'm not sure it's happen for every build and/or on every CI servers. So my question is,  how `--update` pick-up a remote to update the recipe from ?

I believe the solution would be to remove the extra remotes, use `--remote`, or 'just' use the lock graph feature.

### PR DESCRIPTION
**Pull request by danimtb**
Changelog: omit
Docs: omit

#REVISIONS: 1

- [x] Refer to the issue that supports this Pull Request: closes #6104
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

cc/ @AlexandreBossard

### CODE REVIEW

**Comment by danimtb** [comment]
Updated description to enable testing with revisions too.

File: conans/test/functional/remote/multi_remote_test.py [hunk 1]
```diff
@@ -95,6 +95,37 @@ def conan_install_build_flag_test(self):
...
+        client.run("upload Hello0/0.0@lasote/stable --all -r default")
+        sleep(1)  # For timestamp and updates checks
+        self._create(client, "Hello0", "0.0", modifier=" ")
+        client.run("install Hello0/0.0@lasote/stable --build missing")
+        client.run("upload Hello0/0.0@lasote/stable --all -r local")
+        client.run("remove '*' -f")
+
+        client.run("install Hello0/0.0@lasote/stable")
+        self.assertIn("Hello0/0.0@lasote/stable from 'default' - Downloaded", client.out)
+        client.run("install Hello0/0.0@lasote/stable --update")
```
Comment (memsharded) [inline conans/test/functional/remote/multi_remote_test.py]: Yes, seems the test is correct, validating current behavior. Could you please add the check that ``--update -r=local`` will actually update from local repo?

**Comment by memsharded** [comment]
tests broken

**Comment by danimtb** [comment]
The error of the CI seems unrelated to the changes in this PR:

```
ERROR: test_ftp_anonymous (enabled_revisions.conans.test.unittests.client.tools.net_test.ToolsNetTest)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/private/tmp/source/fcb8/py36/enabled_revisions/conans/client/tools/net.py", line 42, in ftp_download

    ftp.retrbinary('RETR ' + filename, f.write)

  File "/Users/jenkins_ci/.pyenv/versions/3.6.5/lib/python3.6/ftplib.py", line 442, in retrbinary

    with self.transfercmd(cmd, rest) as conn:

  File "/Users/jenkins_ci/.pyenv/versions/3.6.5/lib/python3.6/ftplib.py", line 399, in transfercmd

    return self.ntransfercmd(cmd, rest)[0]

  File "/Users/jenkins_ci/.pyenv/versions/3.6.5/lib/python3.6/ftplib.py", line 361, in ntransfercmd

    source_address=self.source_address)

  File "/Users/jenkins_ci/.pyenv/versions/3.6.5/lib/python3.6/socket.py", line 724, in create_connection

    raise err

  File "/Users/jenkins_ci/.pyenv/versions/3.6.5/lib/python3.6/socket.py", line 713, in create_connection

    sock.connect(sa)

ConnectionRefusedError: [Errno 61] Connection refused



During handling of the above exception, another exception occurred:



Traceback (most recent call last):

  File "/private/tmp/source/fcb8/py36/enabled_revisions/conans/test/unittests/client/tools/net_test.py", line 32, in test_ftp_anonymous

    net.ftp_download("speedtest.tele2.net", filename)

  File "/private/tmp/source/fcb8/py36/enabled_revisions/conans/client/tools/net.py", line 48, in ftp_download

    raise ConanException("Error in FTP download from %s\n%s" % (ip, str(e)))

conans.errors.ConanException: Error in FTP download from speedtest.tele2.net

[Errno 61] Connection refused
```

Relaunching the job did not work

**Comment by danimtb** [comment]
Relaunching once again worked. Please @memsharded review

**Comment by danimtb** [comment]
ping

**Pull request merged**